rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -265,97 +447,147 @@ private BlockResult executeInternal(
 
         maintainPrecompiledContractStorageRoots(track, activationConfig.forBlock(block.getNumber()));
 
-        int i = 1;
-        long totalGasUsed = 0;
-        Coin totalPaidFees = Coin.ZERO;
-        List<TransactionReceipt> receipts = new ArrayList<>();
-        List<Transaction> executedTransactions = new ArrayList<>();
-        Set<DataWord> deletedAccounts = new HashSet<>();
+        BlockSharedData blockSharedData = new BlockSharedData(track, programTraceProcessor);
 
-        int txindex = 0;
+        if (!block.isSealed()
+                && block.useParallelTxExecution()
+                && !computeTxPartitioning(block, track, programTraceProcessor, acceptInvalidTransactions, discardInvalidTxs)) {
```

Comment:
The pre-run of transactions sequentially only occurs during mining, and this is correctly detected by block.isSealed(). Nevertheless, a comment would help the reader understand it.

Reply:
I've added some comments to explain better